### PR TITLE
[CFP-337] FormBuilder migration - Transfer details

### PR DESCRIPTION
**Pull request by imtiazAhmd**
#### What
Migrate external_users form step: transfer_details form

#### Ticket

[FormBuilder migration - Transfer details](https://dsdmoj.atlassian.net/browse/CFP-337)

#### TODO

 - [x] Show correct error messages
 - [X] All components replaced with GovUK Formbuilder components
 - [x] Tests passes as normal

### CODE REVIEW

File: app/models/claim/transfer_detail.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class TransferDetail < ApplicationRecord
     include TransferBrainDelegatable
 
     belongs_to :claim, class_name: 'Claim::TransferClaim', inverse_of: :transfer_detail
-    acts_as_gov_uk_date :transfer_date, error_clash_behaviour: :override_with_gov_uk_date_field_error
+    acts_as_gov_uk_date error_clash_behaviour: :override_with_gov_uk_date_field_error
```
Comment (naseberry) [inline app/models/claim/transfer_detail.rb]: As `:transfer_date` is not longer using `gov_uk_date`, is this line still required?

File: app/views/external_users/claims/transfer_fee/_detail_fields.html.haml [hunk 2]
```diff
@@ -1,48 +1,40 @@
...
+        classes: ['js-elected-case'],
+        legend: { text: t('.elected_case'), size: 's' }
+
+      = f.govuk_collection_select :transfer_stage_id,
+        Claim::TransferBrain::TRANSFER_STAGES.values,
+        :id,
+        :description,
+        class: 'js-transfer-stage-id',
+        form_group: { classes: 'fx-autocomplete-wrapper', id: 'cc-transfer-stage' },
+        label: { text: t('.transfer_stage_default_label_text'), class:'js-transfer-stage-label govuk-!-font-weight-bold', for: 'claim_transfer_stage_id' },
```
Comment (naseberry) [inline app/views/external_users/claims/transfer_fee/_detail_fields.html.haml]: The `for` attribute has caused the association to fail with the input.

File: app/views/external_users/claims/transfer_fee/_detail_fields.html.haml [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
+        options: { include_blank: '' }
+
+      = f.govuk_date_field :transfer_date,
+        form_group: {classes: ['js-transfer-date'], id: 'cc-transfer_date' },
```
Comment (naseberry) [inline app/views/external_users/claims/transfer_fee/_detail_fields.html.haml]: Nit / Lint: Add space after opening brackets `{classes`

**Review by naseberry — APPROVED** [review]
Minor issues to resolve.

Code changes look good and I've sanity checked these changes locally.

Waiting for blockers to be resolved before we can proceed.

File: app/models/claim/transfer_detail.rb (hunk 1 above)
Comment (imtiazAhmd) [reply app/models/claim/transfer_detail.rb]: Not required. Removed in latest push

**Pull request merged**